### PR TITLE
Improve check assembly genebuild updates

### PR DESCRIPTION
**Pull request by thomasmaurel**
Adding base_count to assembly index. This is to cover edge cases when a MT sequence or scaffold are added to an assembly after this assembly was released. In these cases the sum of seq_region lengh will change. Also fixing assembly index name in table.sql and test.
Created two new method in Base.pm called check_assembly_update and check_genebuild_update. These methods are now used by the metadataUpdater, report_genomes script and CheckAssemblyGeneset for consistency. We are also doing an extra check on base_count for assembly change to pick up addition of MT or Scaffolds to an existing assembly. GenomeAssemblyInfoAdaptor has been updated to check base_count as well if assembly meta key are the same between two releases. GenomeInfoAdaptor store method will now check base_count when a genome already exist with same assembly meta key, if the base_count if different, the previosu genome will be cleaned up and we will treat this situation as a new assembly (delete previous db as each team will have to re-submit new ones).

### CODE REVIEW

File: sql/patch_13022019_q.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+# patch_13022019_q
+#
+# Title: Allow organism to have same assembly default, accession but different base_count.
+#
+# Description: Remove the assembly_idx on assembly_default and assembly_accession, create new assembly_idx including base_count. The assembly can be the same but the underlying sequence have changed (e.g: new MT, new scaffolds..)
+DROP INDEX assembly_idx on assembly;
```
Comment (marcoooo) [inline sql/patch_13022019_q.sql]: Should this statement be: `DROP INDEX assembly_accession_idx on assembly`. Considering it's a new name you want to apply, you should remove the old one?

Comment (thomasmaurel) [reply sql/patch_13022019_q.sql]: In patch m: https://github.com/Ensembl/ensembl-metadata/blob/release/95/sql/patch_26112018_m.sql. I dropped the `assembly_accession_idx` and replaced it with `assembly_idx`. But I made a mistake in both table.sql and test metadata sql and kept `assembly_accession_idx` by mistake. So this change is also correcting both mistakes. `assembly_idx` is the correct one to drop here.

Comment (marcoooo) [reply sql/patch_13022019_q.sql]: :ok:
